cmd/root.go

```diff
@@ -36,66 +36,76 @@ func mustFlagString(cmd *cobra.Command, name string, required bool) string {
 	return val
 }
 
-type ProviderFunc func(p internal.Provider) error
+type ProviderFunc func(p []internal.Provider) error
 
-func runProvider(logger logger.Logger, url string, dryRun bool, verbose bool, importer string, fn ProviderFunc, nc *nats.Conn) {
+func runProviders(logger logger.Logger, urls []string, dryRun bool, verbose bool, importer string, fn ProviderFunc, nc *nats.Conn) {
 	opts := &provider.ProviderOpts{
 		DryRun:   dryRun,
 		Verbose:  verbose,
 		Importer: importer,
 	}
-	provider, err := provider.NewProviderForURL(logger, url, opts)
-	if err != nil {
-		logger.Error("error creating provider: %s", err)
-		os.Exit(1)
-	}
-	if err := provider.Start(); err != nil {
-		logger.Error("error starting provider: %s", err)
-		os.Exit(1)
-	}
-	if importer != "" {
-		resp, err := http.Get(opts.Importer)
+	providers := []internal.Provider{}
+	for _, url := range urls {
+		provider, err := provider.NewProviderForURL(logger, url, opts)
 		if err != nil {
-			logger.Error("error from importer url request: %s", err)
-			os.Exit(1)
-		}
-		if resp.StatusCode != http.StatusOK {
-			logger.Error("invalid status code from importer url: %s", opts.Importer)
+			logger.Error("error creating provider: %s", err)
 			os.Exit(1)
 		}
-		defer resp.Body.Close()
-
-		gzReader, err := gzip.NewReader(resp.Body)
-		if err != nil {
-			logger.Error("error creating gzip reader: %s", err)
+		if err := provider.Start(); err != nil {
+			logger.Error("error starting provider: %s", err)
 			os.Exit(1)
 		}
-		scanner := bufio.NewScanner(gzReader)
-		const maxCapacity = 1024 * 1024
-		scanner.Buffer(make([]byte, maxCapacity), maxCapacity)
-
-		for scanner.Scan() {
-			data := scanner.Bytes()
+		if importer != "" {
+			resp, err := http.Get(opts.Importer)
+			if err != nil {
+				logger.Error("error from importer url request: %s", err)
+				os.Exit(1)
+			}
+			if resp.StatusCode != http.StatusOK {
+				logger.Error("invalid status code from importer url: %s", opts.Importer)
+				os.Exit(1)
+			}
+			defer resp.Body.Close()
 
-			err = provider.Import(data, nc)
+			gzReader, err := gzip.NewReader(resp.Body)
 			if err != nil {
-				logger.Error("error importing data: %s", err)
+				logger.Error("error creating gzip reader: %s", err)
 				os.Exit(1)
 			}
+			scanner := bufio.NewScanner(gzReader)
+			const maxCapacity = 1024 * 1024
+			scanner.Buffer(make([]byte, maxCapacity), maxCapacity)
+
+			for scanner.Scan() {
+				data := scanner.Bytes()
+
+				err = provider.Import(data, nc)
+				if err != nil {
+					logger.Error("error importing data: %s", err)
+					os.Exit(1)
+				}
+			}
+
+			logger.Info("Importing data instead of streaming")
+			os.Exit(1)
```

Comment:
not sure i understand this abrupt exit with error code. can you add a comment or explain?

Reply:
Yeah- so my logic is now wrong for this part. Previously, I was working on the notion that there was only a single provider allowed, but I'll adjust this logic to fit what it previously did. 

Functionally, I was running off of the idea that after running a mass-import, a company would want to potentially manually confirm that the data successfully imported in their personal databases instead of immediately streaming new data into their database.